torchgeo/trainers/segmentation.py

```diff
@@ -89,20 +90,30 @@ def __init__(self, **kwargs: Any) -> None:
         self.save_hyperparameters()  # type: ignore[operator]
         self.hyperparams = cast(Dict[str, Any], self.hparams)
 
-        self.ignore_zeros = None if kwargs["ignore_zeros"] else 0
+        if not isinstance(kwargs["ignore_zeros"], bool):
+            raise ValueError("ignore_zeros must be a bool")
+        if kwargs["ignore_zeros"] and kwargs["loss"] == "jaccard":
```

Comment:
I specifically warn about this because a user might expect training with jaccard to behave similarly to training with ce or focal loss, however it doesn't